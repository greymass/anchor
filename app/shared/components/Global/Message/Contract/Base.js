// @flow
import React, { Component } from 'react';
import { Accordion, Header, Icon, Message } from 'semantic-ui-react';
import { translate } from 'react-i18next';

class WalletMessageContractBase extends Component<Props> {
  state = { active: false }
  onClick = () => this.setState({ active: !this.state.active });
  render() {
    const { children, t } = this.props;
    const { active } = this.state;
    return (
      <Message color="violet" size="small">
        <Accordion fluid>
          <Accordion.Title active={active} onClick={this.onClick}>
            <Header size="small">
              <Icon name="dropdown" />
              {t('contract_terms')}
            </Header>
          </Accordion.Title>
          <Accordion.Content active={active}>
            {children}
          </Accordion.Content>
        </Accordion>
      </Message>
    );
  }
}

export default translate('contracts')(WalletMessageContractBase);
