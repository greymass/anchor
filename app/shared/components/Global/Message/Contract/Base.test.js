import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalMessageContractBase } from './Base';

const mockProps = {
  children: <React.Fragment />,
  t: jest.fn,
};

describe('GlobalMessageContractBase', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalMessageContractBase {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
