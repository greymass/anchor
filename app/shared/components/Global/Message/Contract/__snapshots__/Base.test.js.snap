// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalMessageContractBase renders correctly 1`] = `
<Message
  color="violet"
  size="small"
>
  <Accordion
    fluid={true}
  >
    <AccordionTitle
      active={false}
      onClick={[Function]}
    >
      <Header
        size="small"
      >
        <Icon
          as="i"
          name="dropdown"
        />
        <Component />
      </Header>
    </AccordionTitle>
    <AccordionContent
      active={false}
    />
  </Accordion>
</Message>
`;
